app/controllers/snippets-management/edit/edit-snippet.js

```diff
@@ -211,12 +212,13 @@ export default class SnippetsManagementEditSnippetController extends Controller
     return this.editorDocument.content !== this.editor?.htmlContent;
   }
 
-  fetchEditorDocument = task(async () => {
-    this._editorDocument = await this.model.currentVersion;
+  currentVersion = trackedFunction(this, async () => {
+    const currentVersion = await this.model.currentVersion;
+    return currentVersion;
   });
 
   get editorDocument() {
-    return this._editorDocument || this.model.currentVersion;
+    return this._editorDocument || this.currentVersion.value;
```

Comment:
I don't really see why `this._editorDocument` is needed. Isn't it always equal to `this.currentVersion`?

Reply:
It was needed before, but with the current changes you are right, I didn't need it